ignite/metrics/mean_average_precision.py

```diff
@@ -0,0 +1,409 @@
+            you want to compute the metric with respect to one of the outputs.
+            By default, metrics require the output as ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+        device: specifies which device updates are accumulated on. Setting the
+            metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
+            non-blocking. By default, CPU.
+
+    Example:
+
+        .. code-block:: python
+
+            import torch
+            from ignite.metrics import MeanAveragePrecision
+
+            # Detection Format:
+            # (id, class, xmin, ymin, width, height, area, crowd, ignore/confidence)
+            # if your bounding box are of the form (xmin, ymin, xmax, ymax) instead
+            # they can be converted using the formulas width = xmax - xmin and
+            # height = ymax - ymin
+
+            # ground truth detections for an image
+            ys = torch.tensor([[1,73,249.1,50.7,375.3,277.2,101928.52,0,0],
+                    [2,73,152.3,50.29,47,312.38, 4097.4,0,0]])
+
+            # predicted detections for the same image
+            y_preds = torch.tensor([[1,73,176.38,50.3,463.62,312.4,144825.6,0,0.328],
+                    [2,73,355.5,111.87,210,93.75,19687.5,0,0.64]])
+
+            mAP = MeanAveragePrecision()
+
+            mAP.update([ys, y_preds])
+
+            mAP.compute()
+
+    .. versionadded:: 0.5.0
+    """
+
+    def __init__(
+        self,
+        object_area_ranges: Optional[Dict[str, Tuple[float, float]]] = None,
+        num_detection_max: int = 100,
+        iou_thresholds: Optional[List[float]] = None,
+        rec_thresholds: Optional[List[float]] = None,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+    ) -> None:
+
+        if object_area_ranges is None:
+            self.object_area_ranges = {
+                "all": (0.0, float("inf")),
+                "small": (0.0, 1024.0),
+                "medium": (1024.0, 9216.0),
+                "large": (9216.0, float("inf")),
+            }
+        else:
+            self.object_area_ranges = object_area_ranges
+            if "all" not in self.object_area_ranges:
+                self.object_area_ranges["all"] = (0, float("inf"))
+
+        self._check_object_area_ranges()
+
+        if num_detection_max < 1:
+            raise ValueError(f"Argument num_detection_max should be a positive integer, got {num_detection_max}")
+
+        self.num_detection_max = num_detection_max
+
+        if iou_thresholds is None:
+            iou_thresholds = [0.5, 0.55, 0.6, 0.65, 0.7, 0.75, 0.8, 0.85, 0.9, 0.95]
+        self.iou_thresholds = torch.tensor(iou_thresholds, device=device)
+
+        if rec_thresholds is None:
+            rec_thresholds = [i / 100 for i in range(101)]
+        self.rec_thresholds = torch.tensor(rec_thresholds, device=device)
+
+        class labels:
```

Comment:
Naming is not PEP8 compliant.

Anyway, the labels would be included in the class. Sorry, it was not a good idea to have it in a specific class. Please, add the labels statically in the class

```python
class MeanAveragePrecision(Metric):
    __id = 0
   __category_id = 1
   ...

   def __init(...):
      ....
```